robot/Cumulus/tests/browser/levels/create_edit_delete_level.robot

```diff
@@ -8,115 +8,102 @@ Library         cumulusci.robotframework.PageObjects
+    Navigate To And Validate Field Value    Source Field           contains       npo02__SmallestAmount__c
+
+
+2 Validate Level Assignment in Batch Job
+    [Documentation]                      Create a contact, edit the smallgift field value to apply a valid
+    ...                                  level by running the batch process
+
+    [tags]                                  W-038641                 feature:Level
     # --------------------------------
-    # Modify the SmallestGift field to change the applied level
+    # Modify the SmallestGift field to allow the level to be applied
     # --------------------------------
-    Scroll Element Into View    text:Donation Totals
-    Click Button       title:Edit Smallest Gift
-    Wait For Locator  record.footer
-    Populate Field          Smallest Gift     2.0
-    Click Record Button     Save
-    Wait Until Loading Is Complete
-    Navigate To And Validate Field Value           Smallest Gift    contains    $2.00    section=Donation Totals
+    Setupdata                               contact                   contact_data=${contact_fields}
+    Set Global Variable                     ${data}
+    Salesforce Update                       Contact                   ${data}[contact][Id]  npo02__SmallestAmount__c=${contact_smallestvalue}
+    Go To Page                              Details
+    ...                                     Contact
+    ...                                     object_id=${data}[contact][Id]
+
+    Navigate To And Validate Field Value    Smallest Gift             contains    $${contact_smallestvalue}
     # --------------------------------
     # Open NPSP Settings and run the Levels batch job
     # --------------------------------
-    Open NPSP Settings      Bulk Data Processes         Level Assignment Batch
-    Click Settings Button    idPanelLvlAssignBatch    Run Batch
-    Wait For Batch To Process        LVL_LevelAssign_BATCH    Completed
+    Open NPSP Settings                      Bulk Data Processes         Level Assignment Batch
+    Click Settings Button                   idPanelLvlAssignBatch       Run Batch
+    Wait For Batch To Process               LVL_LevelAssign_BATCH       Completed
     # --------------------------------
     # Return to the Contact to validate the updated Level field
     # --------------------------------
-    Go To Page    Details    Contact       object_id=${contact_id}
-    Navigate To And Validate Field Value           Level      does not contain         ${level_name}    section=Donation Totals
-    Navigate To And Validate Field Value           Previous Level     contains         ${level_name}    section=Donation Totals
+    Go To Page                              Details
+    ...                                     Contact
+    ...                                     object_id=${data}[contact][Id]
+    Navigate To And Validate Field Value    Level    contains          AutomationLevel
```

Comment:
some steps are missing in this test case. After u verify level, u need to edit smallest gift again and run the batch job to make sure now the previous level is automation level and current level is blank

Reply:
That step is there . After creating the customer, I am updating the small amount value using the API call as The key goal of the test case after talking to manual QE is not if we are able to update the value from UI, but does the level get applied correctly.

Salesforce Update Contact ${data}[contact][Id] npo02__SmallestAmount__c=${contact_smallestvalue}

Reply:
yes I see that u are updating the value, my point was to check if level and previous level fields are updated properly after the change. This test case was added by mike as we were not covering it. May be check with him before if you decide to remove. I am talking in specific to this validation

 Navigate To And Validate Field Value           Level      does not contain         ${level_name}    section=Donation Totals
 Navigate To And Validate Field Value           Previous Level     contains         ${level_name}    section=Donation Totals

Reply:
Got it . I understood it now. Adding the missing part